src/java.base/share/classes/sun/net/www/protocol/http/AuthenticationInfo.java

```diff
@@ -322,39 +301,39 @@ static AuthenticationInfo getServerAuth(URL url, String authenticatorKey) {
      * In this case we do not use the path because the protection space
      * is identified by the host:port:realm only
      */
-    static String getServerAuthKey(URL url, String realm, AuthScheme scheme,
-                                   String authenticatorKey) {
+    static String getServerAuthKey(URL url, String realm, AuthScheme scheme) {
         int port = url.getPort();
         if (port == -1) {
             port = url.getDefaultPort();
         }
         String key = SERVER_AUTHENTICATION + ":" + scheme + ":"
                      + url.getProtocol().toLowerCase()
                      + ":" + url.getHost().toLowerCase()
-                     + ":" + port + ":" + realm
-                     + ";auth=" + authenticatorKey;
+                     + ":" + port + ":" + realm;
         return key;
     }
 
-    private static AuthenticationInfo getCachedServerAuth(String key) {
-        return getAuth(key, null);
+    private static AuthenticationInfo getCachedServerAuth(String key, AuthCacheImpl cache) {
+        return getAuth(key, null, cache);
     }
 
-    static AuthenticationInfo getServerAuth(String key) {
-        if (!serializeAuth) return getCachedServerAuth(key);
-        return requestAuthentication(key, AuthenticationInfo::getCachedServerAuth);
+    static AuthenticationInfo getServerAuth(String key, AuthCacheImpl cache) {
+        if (!serializeAuth) return getCachedServerAuth(key, cache);
+        return requestAuthentication(key, cache, AuthenticationInfo::getCachedServerAuth);
     }
 
 
     /**
      * Return the AuthenticationInfo object from the cache if it's path is
      * a substring of the supplied URLs path.
      */
-    static AuthenticationInfo getAuth(String key, URL url) {
+    static AuthenticationInfo getAuth(String key, URL url, AuthCacheImpl authcache) {
+        // use default cache if none supplied
+        authcache = authcache == null ? defCache : authcache;
```

Comment:
Was this supposed to be a declaration and assignment to a local variable here or are we intentionally overwrting the static `authcache` field here?

Reply:
It sets the local variable (which is intended). It doesn't override the field. But, I agree it's confusing because of the name class. I had changed a number of occurrences of this, but also missed a few. I will rename the local parameter variable.